Source/WebCore/platform/mediastream/gstreamer/RealtimeIncomingSourceGStreamer.cpp

```diff
@@ -158,18 +157,73 @@ void RealtimeIncomingSourceGStreamer::handleDownstreamEvent(GstElement* sink, GR
                 g_object_set(appsrc, "min-latency", minLatency, "max-latency", maxLatency, nullptr);
             });
         }
-        return;
+        return GST_PAD_PROBE_OK;
     }
+    case GST_EVENT_TAG: {
+        GstTagList* eventTags;
+        gst_event_parse_tag(event.get(), &eventTags);
+
+        if (!(GST_IS_TAG_LIST(eventTags)))
+            break;
+
+        if (!GST_IS_TAG_LIST(m_lastTags.get())) {
+            m_lastTags = gst_tag_list_copy(eventTags);
+            break;
+        }
+
+        // baseparse updates the bitrate tags too often, and forwarding all the tag events downstream only because of
+        // small bitrate updates downstream causes excessive overhead. Only send out tag updates if the bitrate tags
+        // have changed by at least 20% or if any other tag has changed.
+        const double bitrateChangeThreshold = 0.2;
+        const char* bitrateTags[] = { GST_TAG_BITRATE, GST_TAG_NOMINAL_BITRATE, GST_TAG_MAXIMUM_BITRATE, GST_TAG_MINIMUM_BITRATE };
```

Comment:
Can span be used here?

Reply:
I simplified the code which made this line unnecessary.